File: Frontend/src/models/attempt.ts
import {AttemptResult} from "./attemptResult";

export class Attempt {
  colors: string[] | undefined;
  attemptResult: AttemptResult | undefined;


  constructor(colors: string[] | undefined, attemptResult: AttemptResult | undefined) {
    this.colors = colors;
    this.attemptResult = attemptResult;
  }
}

File: Backend/Mastermind/Models/Attempt.cs
namespace Mastermind.Models
{
    public class Attempt
    {
        public string[] Colors { get; set; }
        public AttemptResult AttemptResult { get; set; }

        public Attempt(string[] colors, AttemptResult attemptResult)
        {
            Colors = colors;
            AttemptResult = attemptResult;
        }
    }
}